tests/Doctrine/Tests/DBAL/Functional/ConnectionTest.php

```diff
@@ -356,7 +357,36 @@ public function testUserProvidedPDOConnection(): void
 
         $result = $connection->executeQuery('SELECT 1');
 
-        self::assertInstanceOf(ResultStatement::class, $result);
-        self::assertInstanceOf(Result::class, $result);
+        self::assertInstanceOf(ForwardCompatibility\Result::class, $result);
+    }
+
+    public function testResultCompatibilityWhenExecutingQueryWithoutParam(): void
+    {
+        $result = $this->connection->executeQuery('SELECT 1');
+
+        self::assertInstanceOf(ForwardCompatibility\Result::class, $result);
+    }
+
+    public function testResultCompatibilityWhenExecutingQueryWithParams(): void
+    {
+        $result = $this->connection->executeQuery(
+            'SELECT 1, :param',
```

Comment:
You'll need to use something like this to make your tests portable:

https://github.com/doctrine/dbal/blob/4297f5027011027529fdccfa3b1fbbb3090a7d25/tests/Doctrine/Tests/DBAL/Functional/ResultCacheTest.php#L266-L267

Reply:
Thanks ! I couldn't figure out how to make it work with a param, but It work with the basic query. Better than nothing I guess.

Reply:
It's the same: `getDummySelectSQL('1, :param')`.

Reply:
I did try this version, but I still got an error on postgres.